### ISSUE TITLE
[prim_ram] Add reset signal to prim_ram_1p/prim_ram_2p?

### ISSUE THREAD

**Issue by imphil** [lowRISC/opentitan#2494]
prim_ram_1p/prim_ram_2p currently don't have a reset signal input, as they don't need one. (Note that the `_adv` wrappers have a rst signal; this discussion is about the underlying primitives.)

For writing assertions, a reset signal, however, is helpful. Other use cases might come to mind.

I don't have a strong feeling either way, WDYT @tjaychen @msfschaffner @sjgitty @sriyerg?

**Comment by rswarbrick** [comment]
Another option might be to stick the assertions in a "protocol checker" module which you instantiate whenever you actually use the primitive (maybe with a macro or wrapper module; maybe not). That might make it a bit more obvious which bits are part of the actual design and which bits are just for sanity checks.

**Comment by tjaychen** [comment]
i'm not against adding a reset for assertion purposes.
If the reset really ends up being unloaded, in the final round of synthesis
it will be pruned anyways.

is this related to Michael's PR where he just uses a hardcoded 0 for one of
the memory assertions?

**Comment by imphil** [comment]
> is this related to Michael's PR where he just uses a hardcoded 0 for one of
the memory assertions?

Yes, I filed this issue to discuss if we want resets, or not, and to unblock #2487 in the meantime.

**Comment by msfschaffner** [comment]
I am also ok with adding a reset signal to these wrappers for assertion purposes, since as Tim mentioned, it will just be pruned away if unused. The protocol checker would also be an option, but for this one assertion I think we need not necessarily introduce another module...

**Comment by tjaychen** [comment]
This is now related to #12602 
In addition to bringing in a reset, in general it would be nice to have an init function so that software would not need to do it manually in the future.

**Comment by msfschaffner** [comment]
Related issue [#2263](https://github.com/lowRISC/opentitan/issues/2263)

**Comment by rswarbrick** [comment]
The single-port version now has a reset signal (since 7648c5b1898). `prim_ram_2p` still doesn't.

I don't think there's a discussion to have here (especially since this has been silent since 2022). Closing.
